sky/clouds/service_catalog/images/README.md

```diff
@@ -10,42 +10,58 @@ packer init plugins.pkr.hcl
 3. Setup cloud credentials
 
 ## Generate Images
-```bash
-export CLOUD=gcp    # Update this
-export TYPE=gpu    # Update this
-export IMAGE=skypilot-${CLOUD}-${TYPE}-ubuntu
-packer build ${IMAGE}.pkr.hcl
-```
-You will see the image ID after the build is complete.
-
-FYI time to packer build an image:
-
+FYI time to packer build images:
 | Cloud | Type | Approx. Time |
 |-------|------|------------------------|
 | AWS   | GPU  | 15 min          |
 | AWS   | CPU  | 10 min          |
 | GCP   | GPU  | 16 min          |
 | GCP   | CPU  | 5 min          |
+| Azure | GPU  | 35 min          |
+| Azure | CPU  | 25 min          |
 
 ### GCP
+1. Build a single global image.
+```bash
+export TYPE=gpu  # Update this
+export IMAGE=skypilot-gcp-${TYPE}-ubuntu
+packer build ${IMAGE}.pkr.hcl
+```
+2. Make the image public
 ```bash
-export IMAGE_NAME=skypilot-gcp-cpu-ubuntu-20241011003407  # Update this
-
 # Make image public
+export IMAGE_NAME=skypilot-gcp-cpu-ubuntu-xxx  # Update this
 export IMAGE_ID=projects/sky-dev-465/global/images/${IMAGE_NAME}
 gcloud compute images add-iam-policy-binding ${IMAGE_NAME} --member='allAuthenticatedUsers' --role='roles/compute.imageUser'
 ```
 
 ### AWS
-1. Generate images for all regions
+1. Generate the source image for a single region.
+```bash
+export TYPE=gpu  # Update this
+export IMAGE=skypilot-aws-${TYPE}-ubuntu
+packer build ${IMAGE}.pkr.hcl
+```
+2. Copy images to all regions
 ```bash
 export IMAGE_ID=ami-0b31b24524afa8e47   # Update this
-
 python aws_utils/image_gen.py --image-id ${IMAGE_ID} --processor ${TYPE}
 ```
-2. Add fallback images if any region failed \
+3. Add fallback images if any region failed \
 Look for "NEED_FALLBACK" in the output `images.csv` and edit. (You can use public [ubuntu images](https://cloud-images.ubuntu.com/locator/ec2/) as fallback.)
 
+### Azure
+1. Generate a client secret for packer [here](https://portal.azure.com/?feature.msaljs=true#view/Microsoft_AAD_RegisteredApps/ApplicationMenuBlade/~/Credentials/appId/1d249f23-c22e-4d02-b62b-a6827bd113fe/isMSAApp~/false).
+```bash
+export SECRET=xxxxxx  # Update this
+```
+2. Build and copy images for all regions
+```bash
+export VM_GENERATION=2  # Update this
```

Comment:
I assume we will need both generation? Should we mention it in the instruction here?